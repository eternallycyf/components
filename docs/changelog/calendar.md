---
title: Calendar
nav:
  title: changelog
  order: 999
tocDepth: 2
---

## Calendar

<embed src="../../packages/calendar/CHANGELOG.md"></embed>
