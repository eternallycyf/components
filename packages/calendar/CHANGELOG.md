# Changelog

## [Version&nbsp;1.3.0](https://github.com/eternallycyf/components/compare/@ims-view/calendar@1.2.1...@ims-view/calendar@1.3.0)

<sup>Released on **2024-05-05**</sup>

#### ✨ 新特性

- Add umd build.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add umd build ([7df21fa](https://github.com/eternallycyf/components/commit/7df21fa))

</details>

### [Version&nbsp;1.2.1](https://github.com/eternallycyf/components/compare/@ims-view/calendar@1.2.0...@ims-view/calendar@1.2.1)

<sup>Released on **2024-04-28**</sup>

#### 🐛 修复

- Keep react version.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- Keep react version ([fee7e8d](https://github.com/eternallycyf/components/commit/fee7e8d))

</details>

## [Version&nbsp;1.2.0](https://github.com/eternallycyf/components/compare/@ims-view/calendar@1.1.1...@ims-view/calendar@1.2.0)

<sup>Released on **2024-04-27**</sup>

#### ✨ 新特性

- Add Icon component.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add Icon component ([33e4af2](https://github.com/eternallycyf/components/commit/33e4af2))

</details>

### [Version&nbsp;1.1.1](https://github.com/eternallycyf/components/compare/@ims-view/calendar@1.1.0...@ims-view/calendar@1.1.1)

<sup>Released on **2024-02-09**</sup>

#### 🐛 修复

- **@ims-view/calendar**: Add dependencies.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **@ims-view/calendar**: Add dependencies ([f8baa4c](https://github.com/eternallycyf/components/commit/f8baa4c))

</details>

## [Version&nbsp;1.1.0](https://github.com/eternallycyf/components/compare/@ims-view/calendar@1.0.0...@ims-view/calendar@1.1.0)

<sup>Released on **2024-02-09**</sup>

#### ✨ 新特性

- **@ims-view/calendar**: Add Calendar.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/calendar**: Add Calendar ([14fff6d](https://github.com/eternallycyf/components/commit/14fff6d))

</details>

## Version&nbsp;1.0.0

<sup>Released on **2024-02-09**</sup>

#### ✨ 新特性

- **@ims-view/calendar**: Add MiniCalendar.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/calendar**: Add MiniCalendar ([6be604f](https://github.com/eternallycyf/components/commit/6be604f))

</details>
