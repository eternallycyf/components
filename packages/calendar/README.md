# @ims-view/calendar

[![NPM version][version-image]][version-url] [![NPM downloads][download-image]][download-url]

## License

[MIT](../../LICENSE) Â® eternallycyf

<!-- npm url -->

[version-image]: http://img.shields.io/npm/v/@ims-view/calendar.svg?color=deepgreen&label=latest
[version-url]: http://npmjs.org/package/@ims-view/calendar
[download-image]: https://img.shields.io/npm/dm/@ims-view/calendar.svg
[download-url]: https://npmjs.org/package/@ims-view/calendar

<!-- repo url -->

- https://github.com/eternallycyf/components
