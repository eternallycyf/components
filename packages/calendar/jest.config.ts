import { Config } from '@umijs/test';
import base from '../../jest.config.base';

const packageName = '@ims-view/calendar';

const root = '<rootDir>/packages/calendar';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
};

export default config;
