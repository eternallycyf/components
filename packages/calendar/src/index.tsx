export { default as MiniCalendar } from './Calendar';
export * from './Calendar/interface';
