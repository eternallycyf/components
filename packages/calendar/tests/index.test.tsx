import { render } from '@testing-library/react';

import { Calendar } from '@ims-view/calendar';
import React from 'react';

test('Calendar', () => {
  const { container } = render(<Calendar />);

  expect(container).toMatchSnapshot();
});
