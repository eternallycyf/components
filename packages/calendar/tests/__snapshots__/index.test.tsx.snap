// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar 1`] = `
<div>
  <div>
    <div
      class="calendar"
    >
      <div
        class="header"
      >
        <button
          type="button"
        >
          &lt;
        </button>
        <div>
          2024
          年
          八月
        </div>
        <button
          type="button"
        >
          &gt;
        </button>
      </div>
      <div
        class="days"
      >
        <div
          class="day"
        >
          日
        </div>
        <div
          class="day"
        >
          一
        </div>
        <div
          class="day"
        >
          二
        </div>
        <div
          class="day"
        >
          三
        </div>
        <div
          class="day"
        >
          四
        </div>
        <div
          class="day"
        >
          五
        </div>
        <div
          class="day"
        >
          六
        </div>
        <div
          class="empty"
        />
        <div
          class="empty"
        />
        <div
          class="empty"
        />
        <div
          class="empty"
        />
        <div
          class="day"
        >
          1
        </div>
        <div
          class="day"
        >
          2
        </div>
        <div
          class="day"
        >
          3
        </div>
        <div
          class="day"
        >
          4
        </div>
        <div
          class="day"
        >
          5
        </div>
        <div
          class="day"
        >
          6
        </div>
        <div
          class="day"
        >
          7
        </div>
        <div
          class="day"
        >
          8
        </div>
        <div
          class="day"
        >
          9
        </div>
        <div
          class="day"
        >
          10
        </div>
        <div
          class="day"
        >
          11
        </div>
        <div
          class="day"
        >
          12
        </div>
        <div
          class="day"
        >
          13
        </div>
        <div
          class="day"
        >
          14
        </div>
        <div
          class="day selected"
        >
          15
        </div>
        <div
          class="day"
        >
          16
        </div>
        <div
          class="day"
        >
          17
        </div>
        <div
          class="day"
        >
          18
        </div>
        <div
          class="day"
        >
          19
        </div>
        <div
          class="day"
        >
          20
        </div>
        <div
          class="day"
        >
          21
        </div>
        <div
          class="day"
        >
          22
        </div>
        <div
          class="day"
        >
          23
        </div>
        <div
          class="day"
        >
          24
        </div>
        <div
          class="day"
        >
          25
        </div>
        <div
          class="day"
        >
          26
        </div>
        <div
          class="day"
        >
          27
        </div>
        <div
          class="day"
        >
          28
        </div>
        <div
          class="day"
        >
          29
        </div>
        <div
          class="day"
        >
          30
        </div>
        <div
          class="day"
        >
          31
        </div>
      </div>
    </div>
  </div>
</div>
`;
